src/main.rs

```diff
@@ -343,34 +344,42 @@ fn run(
         let mut sx_old = 0;
         let mut sy_old = 0;
 
-        move || loop {
-            if event::poll(update_tick_rate).unwrap() {
+        move |next_reload: Instant, next_diff: Option<Instant>| loop {
+            let deadline = if let Some(next) = next_diff {
+                next.min(next_reload)
+                    .saturating_duration_since(Instant::now())
+            } else {
+                next_reload.saturating_duration_since(Instant::now())
+            };
+            if event::poll(deadline).unwrap() {
                 match event::read().unwrap() {
                     CEvent::Key(key) => return Event::Input(key),
                     CEvent::Mouse(_) => (),
                     CEvent::Resize(sx, sy) => {
                         if sx != sx_old || sy != sy_old {
                             sx_old = sx;
                             sy_old = sy;
-                            return Event::Tick;
+                            return Event::Update;
                         }
                     }
                 }
                 continue;
             }
-
             return Event::Update;
         }
     };
 
     terminal.clear()?;
 
+    let mut next_diff_update = None;
```

Comment:
Since `next_event` is closure, you could move these up before its definition and access them by capturing references instead of passing them as arguments. (This would require putting them inside `Cell<_>` but this should not be an issue for `Instant`.)

Reply:
Mh, problems with the borrow checker when trying this. I probably should use my first `Cell` with a simpler example.

Reply:
Done.